core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -245,20 +245,33 @@ function createRowMetadataForColumn(colKey, data, checkboxId) {
 function reselectCheckboxesBasedOnTaskTableState() {
     var allChecked = true;
     var taskSummaryMetricsTableCurrentFilteredArray = taskSummaryMetricsTableCurrentStateArray.slice();
-    if (typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0) {
-        for (var k = 0; k < optionalColumns.length; k++) {
-            if (taskTableSelector.column(optionalColumns[k]).visible()) {
-                $("#box-"+optionalColumns[k]).prop('checked', true);
-                taskSummaryMetricsTableCurrentStateArray.push(taskSummaryMetricsTableArray.filter(row => (row.checkboxId).toString() == optionalColumns[k])[0]);
-                taskSummaryMetricsTableCurrentFilteredArray = taskSummaryMetricsTableCurrentStateArray.slice();
-            } else {
-                allChecked = false;
+    if ((typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0) || typeof executorSummaryTableSelector !== 'undefined') {
+        if (typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0) {
```

Comment:
The condtion `typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0`  repeats twice here. 
Can we separate the condition `typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0` and `typeof executorSummaryTableSelector !== 'undefined'` ?

Reply:
> The condtion `typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0` repeats twice here.
> Can we separate the condition `typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0` and `typeof executorSummaryTableSelector !== 'undefined'` ?

Updated